packages/dds/tree/src/core/tree/anchorSet.ts

```diff
@@ -99,6 +101,46 @@ export interface AnchorEvents {
 	 * It's possible nodes removed from the tree will be kept indefinably, and thus never trigger this event, or they may be discarded immediately.
 	 */
 	afterDelete(anchor: AnchorNode): void;
+
+	/**
+	 * A change to what children the node has.
+	 */
+	childrenChange(anchor: AnchorNode): void;
+
+	/**
+	 * Something in this tree changed.
+	 * Called on every parent (transitively) that changed.
```

Comment:
This makes me think consumers of these events may want to option to return some value indicating whether they want the even to keep bubbling up or not.

Reply:
that kind of thing is up to the binder API (which does not exist yet but will use these events). This API's job is to be efficient as possible while providing enough entry points that is possible for binder to work efficiently. Calling the event while traversing is super cheap, but doing bubbling would require more complex logic, and be a bit odd semantically since there may be many events in the subtree which could trigger this, and once we have subdeltsa they will all be included in the delta we provide to this event even if you tried to stop a subset from bubbling.

Note that in the case where you just want event bubbling, binder probably wouldn't register events on the parents: it could do the bubbling by walking up itself. It will use these events this behavior for when it wants to use a single registration to get events for changes to some big subtree without making lots of registrations (and thus be able to still get events, but not have to make anchors for every node).

Reply:
Sounds good